pkg/gadgets/run/tracer/tracer.go

```diff
@@ -376,6 +376,46 @@ func (t *Tracer) attachProgram(gadgetCtx gadgets.GadgetContext, p *ebpf.ProgramS
 		case strings.HasPrefix(p.SectionName, "kretprobe/"):
 			logger.Debugf("Attaching kretprobe %q to %q", p.Name, p.AttachTo)
 			return link.Kretprobe(p.AttachTo, prog, nil)
+		case strings.HasPrefix(p.SectionName, "uprobe/"):
+			captureHost := false
+			for _, container := range t.containers {
+				if container.Pid == 1 {
+					captureHost = true
+				}
+			}
+			if !captureHost {
+				return nil, fmt.Errorf("uprobe can only be used with --host at this moment")
```

Comment:
The test for `--host` is not ideal but this is ok for me since the next PR will add support for tracing containers.

Reply:
Yes, the handling of `--host` here doesn't align with its original intent. According to the documentation of IG, when providing the `--host`, it's not only necessary to track processes on the host machine but also processes within containers.
However, at present, we're unable to implement tracing within containers. So we force users to specify `--host`, and we are only tracking processes outside of containers.

Reply:
The next PR will come soon, so we can refactor it later.